src/thirtytwo/Window.cs

```diff
@@ -283,8 +290,29 @@ private LRESULT WindowProcedureInternal(HWND window, uint message, WPARAM wParam
 
         // Ensure we're not collected while we're processing a message.
         GC.KeepAlive(this);
+        
+        // Re-throw exception if one was caught
```

Comment:
Rethrowing here would let other code run after the invalid state. As stated above, only a finally is needed to accomplish this fix.